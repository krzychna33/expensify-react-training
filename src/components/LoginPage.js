import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../actions/auth';

export class LoginPage extends React.Component {
    render(){
        return (
            <div>
                <h1>Expensify app</h1>
                <h4>Please log in.</h4>
                <button onClick={this.props.startLogin}>Login</button>
            </div>
        )
    }
};

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);