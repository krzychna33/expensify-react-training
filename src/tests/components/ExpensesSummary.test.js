import React from 'react'
import { shallow } from 'enzyme'
import {ExpensesSummary} from '../../components/ExpensesSummary';

test('Should render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={1000}/>);
    expect(wrapper).toMatchSnapshot();
})

test('Should render ExpensesSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={5} expensesTotal={10000}/>);
    expect(wrapper).toMatchSnapshot();
})