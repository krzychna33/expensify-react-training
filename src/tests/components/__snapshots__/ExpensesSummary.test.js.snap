// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary with 1 expense 1`] = `
<div>
  <p>
    Currently viewing 
    1
     
    expense
    , totalling 
    $ 10.00 
  </p>
</div>
`;

exports[`Should render ExpensesSummary with multiple expenses 1`] = `
<div>
  <p>
    Currently viewing 
    5
     
    expenses
    , totalling 
    $ 100.00 
  </p>
</div>
`;
